armi/physics/neutronics/globalFlux/globalFluxInterface.py

```diff
@@ -1313,6 +1313,82 @@ def computeDpaRate(mgFlux, dpaXs):
     return dpaPerSecond
 
 
+def calcReactionRatesBlockList(objList, keff, xsNucDict):
+    r"""
+    Compute 1-group reaction rates for the objcects in objList (usually a block).
+
+    Parameters
+    ----------
+    objList : List[Block]
+        The list of objects to compute reaction rates on. Notionally this could be upgraded to be
+        any kind of ArmiObject but with params defined as they are it currently is only
+        implemented for a block.
+
+    keff : float
+        The keff of the core. This is required to get the neutron production rate correct
+        via the neutron balance statement (since nuSigF has a 1/keff term).
+
+    xsNucDict: Dict[str, XSNuclide]
+        Microscopic cross sections to use in computing the reaction rates.
+
+    .. impl:: Return the reaction rates for a given ArmiObject
+        :id: I_ARMI_FLUX_RX_RATES_BY_XS_ID
+        :implements: R_ARMI_FLUX_RX_RATES
+
+        This is an alternative implementation of :need:`I_ARMI_FLUX_RX_RATES` that
+        is more efficient when computing reaction rates for a large set of blocks
+        that share a common set of microscopic cross sections.
+
+        Fore more detail on the reation rate calculations, see :need:`I_ARMI_FLUX_RX_RATES`.
+    """
+    for obj in objList:
+        rate = {}
+        for simple in RX_PARAM_NAMES:
+            rate[simple] = 0.0
+
+        numberDensities = obj.getNumberDensities()
+        try:
+            mgFlux = np.array(obj.getMgFlux())
+        except TypeError:
+            continue
```

Comment:
Should we allow for this function to silently continue if an object without flux is passed in? The explicit purpose of this function is to calculate reaction rates, so passing an object without flux is just a fundamentally wrong premise.

A hard error feels appropriate to me here, unless there is a common case that I am not aware of.

Reply:
This is more copy/paste:

![image](https://github.com/user-attachments/assets/2b6d5217-c3e0-4272-b177-a9a962b95386)

Based on the comment, it seems like the thinking goes that if an object without flux is passed in, you can just skip calculating the reaction rates on that object. You don't have to crash the whole run.

Reply:
I don't know whether we're ever actually hitting this exception. It might be good to remove it if possible, but that could probably be a separate PR where it is also removed from the original function I copied it out of.